.travis.yml

```diff
@@ -17,6 +17,7 @@ before_script:
 script:
   - mkdir -p build/logs
   - vendor/bin/phpunit --coverage-clover build/logs/clover.xml
+  - vendor/bin/psalm --show-info=false --no-progress
```

Comment:
Right now, this is run after every single build in the matrix, which is a waste of resources. There should be a job dedicated to running the static analyzer, and no PHPUnit tests. See for example: https://travis-ci.org/doctrine/dbal

Reply:
Good idea. Done. I also moved code coverage report to this "only once" section.

Reply:
Careful: coverage has to be performed on every job, as different calculators will be covered by different tests!

Also, please don't use a "Smoke testing" stage, this can belong to the main test stage. Doctrine uses smoke testing to avoid running the main test suite if the smoke test fails, but this is not what I had in mind here.